libdnf/utils/temp.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef LIBDNF_UTILS_TEMP_HPP
+#define LIBDNF_UTILS_TEMP_HPP
+
+
+#include <filesystem>
+#include <vector>
+#include <string>
+
+
+namespace libdnf::utils {
+
+
+/// Object that creates and holds a temp directory.
+/// The directory gets removed when the object is deleted.
+class TempDir {
+public:
+    explicit TempDir(const std::string & prefix);
+    // create a temp directory with specified subdirs
+    explicit TempDir(const std::string & prefix, std::vector<std::string> subdirs);
```

Comment:
No need to copy the vector.
`explicit TempDir(const std::string & prefix, const std::vector<std::string> & subdirs);`

Reply:
Fixed.